common/hooks/pre-pkg/03-rewrite-python-shebang.sh

```diff
@@ -13,15 +13,28 @@ hook() {
 	fi
 
 	if [ -n "$pyver" ]; then
-		shebang="#!/usr/bin/python${pyver%.*}"
+		default_shebang="#!/usr/bin/python${pyver%.*}"
 	fi
 
 	find "${PKGDESTDIR}" -type f -print0 | \
 		while IFS= read -r -d '' file; do
 			[ ! -s "$file" ] && continue
-			[ -z "$(sed -n -E -e 2q -e '/^#!.*([[:space:]]|\/)python([0-9]\.[0-9])?([[:space:]]+|$)/p' "$file")" ] && continue
-			[ -n "$shebang" ] || msg_error "cannot convert shebang, set python_version\n"
-			echo "   Shebang converted to '$shebang': ${file#$PKGDESTDIR}"
+
+			pyinterp=$(sed -n -E -e 2q -e 's@^#!.*([[:space:]]|/)(python)([0-9](\.[0-9]+)?)?([[:space:]]+|$)@\2\3@p' "$file")
```

Comment:
I think the whole
```sed
s@^#!.*([[:space:]]|/)(python)([0-9](\.[0-9]+)?)?([[:space:]]+|$)@\2\3@p
```

can be simplified to:
```sed
s@^#!.*([[:space:]]|/)(python[0-9.]+)?([[:space:]]+|$)@\2\3@p
```

Note that I still keep the space after `python*`
because it's nice to keep whatever parameter there :)
I didn't put much thought into that, though.

Reply:
Anyway, is that `?` necessary? I think it isn't

Reply:
Your simplification is an improvement, but it would match things like `python..` or `python.7` which thwarts the remaining version detection logic. I think `(python([0-9][0-9.]*)?)` is a good compromise to ignore strange corner cases that don't fit the automatic remapping design.